libraries/AP_Baro/AP_Baro.cpp

```diff
@@ -617,3 +618,44 @@ void AP_Baro::set_pressure_correction(uint8_t instance, float p_correction)
         sensors[instance].p_correction = p_correction;
     }
 }
+
+/*
+ * register new baro if there is space for it
+ */
+bool AP_Baro::register_uavcan_baro(uint8_t mgr, uint8_t node)
+{
+#if HAL_WITH_UAVCAN
+    // Check if this node in network is already registered
+    for (uint8_t i = 0; i < _num_drivers; i++) {
+        if (sensors[i].can_manager == mgr && sensors[i].uavcan_node == node) {
+            return true;
+        }
+    }
+
+    // If there is place left - allocate one UAVCAN based baro
+    if (_num_drivers < BARO_MAX_DRIVERS && _num_sensors < BARO_MAX_INSTANCES)
+    {
+        if ((AP_BoardConfig_CAN::get_can_enable() != 0) && (hal.can_mgr[mgr] != nullptr))
+        {
+            drivers[_num_drivers] = new AP_Baro_UAVCAN(*this);
+
+            if (((AP_Baro_UAVCAN *) drivers[_num_drivers])->register_uavcan_baro(mgr, node)) {
```

Comment:
For baro we do not have 'type' field like for GPS. In UAVCAN you can not probe it. So only way to know if it is there (or even can be there at all) is to wait for some time for any message to arrive from it. This delay was there previously and I changed it to hot-plug method.